### DISCUSSION TITLE
Problems with Orange Upgrade

### DISCUSSION THREAD

**Discussion by xanant** [biolab/orange3#5389] [Q&A]
HI, I'm having trouble upgrading from Orange3 vers. 3.26.0 to the latest vers. 3.28.0 using Anaconda. When I try to upgrade by Anaconda Navigator the computer starts but waits very long and does not upgrade. Furthermore, if I try to install the Explain 0.3.0 add-on, in Orange3 vers 3.26.0, the installation stops telling me that Orange3 vers. 3.28.0 must be installed first. It seems like a vicious circle. Below is my configuration to understand if there is any inconsistency:
MacOS El Capitan v. 10.11.6
Orange3 v. 3.26.0
Anaconda Navigator v. 1.10.0
Python3 v. 3.9.4 (but also present Python 2.7)
Can you help me out?
Thanks

**Comment by ajdapretnar** [comment]
My first assumption would be that you do not have the right channel set in Anaconda. Could you try [this](https://orangedatamining.com/blog/2020/2020-02-24-anaconda-navigator/) and see if it works then?

**Comment by xanant** [comment]
Hi Ajda, thanks for the reply. I had just followed your instructions by creating the Orange environment. I redid the procedure again and I'm afraid I have made more trouble as I no longer find the possibility of updating from Anaconda. I attach some screenshots for clarification. When I open Orange it warns me about the possibility of updating, but using Anaconda I don't understand if I have to use the terminal anyway with 'conda config --add channels conda-forge' and 'conda install orange3' (?!) because it seems that I can't use others methods to update. Where am I wrong? Sorry for disturbing.

**Reply by ajdapretnar** [reply]
So if you go to the Orange environment in Anaconda, where you have set the priority channel to conda-forge, you cannot update anymore? Could you take a screenshot of it?

**Comment by xanant** [comment]
I make a lot of confusion. Before installing the latest version of Anaconda Navigator everything went smoothly. I was using Anaconda and Orange proceeding with updates without any problem (no needing an Orange Environment). Now (after created Orange Environment) on the Anaconda Home screen (unlike the one in your instructions) I find what you see in screenshot 3A and so also in Environments in screenshot 3B: Orange3 only has version 3.26.0 and not the later ones nor the possibility to update. I think I have not understood the functions of channels and Environments. Excuse my ignorance but I don't know how to proceed further.

2 screenshots to highlight the situation in which I find myself.

**Comment by xanant** [comment]
Hi Ajda

about my problem I suppose I understand what it consists of. My MacOS ElCapitan 10.11.6 does not support Orange3 vers. 3.28.0 (as shown in the  attached error screenshot) and I can only use vers. 3.26.0. This is my conclusion but I would appreciate your reply for any confirmation. Upgrading the MacOs involves a series of changes to the programs already installed and I have to think carefully if it suits me. Thanks

**Reply by ajdapretnar** [reply]
@xanant You have still not provided any screenshots. Please use Github directly, not the reply function in mail (I suppose this is the issue of attaching screenshots).

**Comment by xanant** [comment]
Hi Ajda
forgive.. this is the screenshot reporting the error i get where referring to the macOS higher than mine. From here I deduce my guess. 
![Error Anaconda-Orange3](https://user-images.githubusercontent.com/63248513/114373287-bd5d6380-9b82-11eb-9886-6769a7b36aff.png)

**Comment by xanant** [comment]
Hi
I would like to suggest to insert, in the Orange3 standalone 'DMG package' for Mac, a note specifying the OS needed for the download version. Mac packages not suitable for a particular OS, in general, warn and do not allow installation immediately. Or specify which Orange version for which MacOS version. It would be useful. 
Best Regards
